src/operation.ts

```diff
@@ -0,0 +1,27 @@
+import { Operation as ApolloOperation } from '@apollo/client/core';
+import {
+  getOperationAST,
+  OperationDefinitionNode,
+  OperationTypeNode,
+} from 'graphql';
+
+export interface Operation extends ApolloOperation {
+  operationType: OperationTypeNode;
+  operationDefinition: OperationDefinitionNode;
+}
+
+export function makeOperation(apolloOperation: ApolloOperation): Operation {
+  const { query, operationName } = apolloOperation;
+
+  // We know this can not be null because Apollo validates the document/operationName before
```

Comment:
I am actually not entirely sure if they do, will try it out soon.

Reply:
I tried it out, they actually don't even allow multiple operations within a doc.